### ISSUE TITLE
bug(cli): --safe-to-edit silently processes directories as files; --agent-workflow rejects them with wrong error + exit 0

### ISSUE THREAD

**Issue by aimasteracc** [aimasteracc/tree-sitter-analyzer#1002]
From dogfood session on `develop` @ 35d0dac0 (2026-06-16). Path/input-validation inconsistency across safe-to-edit, agent-workflow, and exit-code semantics.

**Symptoms (three related findings)**

### Finding 1: `--safe-to-edit <directory>` silently processes the directory as a file

```
$ uv run python -m tree_sitter_analyzer --safe-to-edit "tree_sitter_analyzer/cli" --format json
{
  "success": true,
  "verdict": "UNSAFE",
  "file_path": "tree_sitter_analyzer/cli",      ← a DIRECTORY
  "downstream_count": 248,                       ← bogus for a dir
  "language": null,
  "error": null,
  ...
}
EXIT: 0
```

The tool does not reject the directory input. It returns a full risk envelope with a suspicious `downstream_count` (248) and `language: null`. An agent consuming this has no idea the input was invalid.

Compare with `--agent-workflow` which DOES reject directories — but with the **wrong message**:

### Finding 2: `--agent-workflow <directory>` rejects directories but with a misleading error

```
$ uv run python -m tree_sitter_analyzer --agent-workflow "tree_sitter_analyzer/cli" --format json
{
  "success": false,
  "verdict": "ERROR",
  "error": "target_path 'tree_sitter_analyzer/cli' is outside the project root",   ← WRONG
  ...
}
EXIT: 0    ← also wrong: verdict=ERROR but exits 0
```

The path is **inside** the project root. The actual problem is that it's a directory, not a file. The error message lies about why.

### Finding 3: Exit-code semantics are inconsistent across input-validation errors

```
safe-to-edit  <directory>     EXIT: 0   ← should be non-zero (silent dir-as-file)
agent-workflow <directory>    EXIT: 0   ← should be non-zero (verdict=ERROR)
codegraph-navigate ""         EXIT: 1   ← good
symbol-lineage   ""           EXIT: 1   ← good
```

Two tools that should reject bad input either succeed silently (safe-to-edit) or exit 0 despite an ERROR verdict (agent-workflow). The other two correctly exit non-zero.

**Why this matters**

1. **Agents gating on exit code miss the agent-workflow error.** A bash-style `if uv run ... --agent-workflow "$path"; then ...` proceeds to the next step even though the workflow command failed.
2. **Agents gating on `success`/`verdict` get contradictory signals from safe-to-edit.** The envelope says success=true / verdict=UNSAFE on a directory — an agent might trust the `downstream_count` and split work based on bogus data.
3. **Inconsistent error messages erode agent trust.** "outside the project root" is gaslighting when the path is obviously inside it. An agent that learns to mistrust TSA error messages will stop calling the CLI.
4. **AGENTS.md anti-pattern #1 risk** (no skip-and-paper-over): silently processing a directory as a file is papering over the validation gap.

**Fix direction**

1. **Centralize path validation.** Both `--safe-to-edit` and `--agent-workflow` (and every other file-path-consuming flag) should go through one helper:

   ```python
   def validate_file_target(path: str, project_root: Path) -> Path:
       resolved = (project_root / path).resolve()
       if not _is_within(project_root, resolved):
           raise TargetError(f"path '{path}' is outside the project root")
       if not resolved.exists():
           raise TargetError(f"path '{path}' does not exist")
       if resolved.is_dir():
           raise TargetError(f"path '{path}' is a directory, expected a file")
       return resolved
   ```

2. **Map verdict → exit code consistently.** Document and enforce:

   | verdict | exit code |
   |---|---|
   | SAFE, INFO, REVIEW | 0 |
   | WARN | 0 (with stderr note) |
   | UNSAFE | 0 (UNSAFE is a result, not an error) |
   | NOT_FOUND | 1 |
   | ERROR | 2 |

   Today `agent-workflow` returns verdict=ERROR with exit 0, which violates this.

3. **Add a path-validation contract test.** Pin the behavior for both file and directory inputs across `--safe-to-edit`, `--agent-workflow`, `--smart-context`, `--file-health`, `--refactor`.

**Test gap**

```python
@pytest.mark.parametrize("tool_flag", ["--safe-to-edit", "--agent-workflow", "--smart-context"])
def test_file_target_rejects_directory_with_clear_error(tool_flag, cli_runner, tmp_project):
    dir_path = tmp_project / "src"
    dir_path.mkdir()
    result = cli_runner([tool_flag, str(dir_path), "--format", "json"])
    assert result.exitcode != 0, f"{tool_flag} must reject directories with non-zero exit"
    assert "directory" in result.stderr.lower() or "directory" in result.stdout.lower()
    assert "outside the project root" not in result.stdout  # message must be accurate
```

**Priority**

P2 — input-validation is the first agent-facing surface; inconsistency here cascades into all downstream tool calls.

**Comment by aimasteracc** [comment]
Resolved by #1008 (merged).
